fili-core/src/main/java/com/yahoo/bard/webservice/data/ResultSetSchema.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data;
+
+import com.yahoo.bard.webservice.druid.model.query.Granularity;
+import com.yahoo.bard.webservice.table.BaseSchema;
+import com.yahoo.bard.webservice.table.Column;
+import com.yahoo.bard.webservice.table.Schema;
+
+import java.util.LinkedHashSet;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * The schema for a result set.
+ */
+public class ResultSetSchema extends BaseSchema implements Schema {
```

Comment:
I'm switching to an Iterable in LinkedHashSet out idiom
One of the libraries we use supports that and it makes a lot of sense because it unambiguously commuicates that we're set-ifying, and the Iterable makes it clear that we're concerned with order while being polymorphic across sources.